src/hf_hydrodata/gridded.py

```diff
@@ -1909,11 +1963,57 @@ def _read_and_filter_tiff_files(
     data_ds = xr.open_dataset(file_path)
     data_da = data_ds[variable]
     da_indexers = _create_da_indexer(options, entry, data_ds, data_da, file_path)
-    data_da = data_da.isel(da_indexers)
-    data = data_da.to_numpy()
+    if _flip_da_indexers(entry, da_indexers):
+        # Select data with the flipped da_indexers and flip the result data
+        data_da = data_da.isel(da_indexers)
+        if len(data_da.shape) == 3:
+            data = np.flip(data_da.to_numpy(), 1)
+        elif len(data_da.shape) == 2:
+            data = np.flip(data_da.to_numpy(), 0)
+    else:
+        # Select the data and do not flip the result
+        data_da = data_da.isel(da_indexers)
+        data = data_da.to_numpy()
     return data
 
 
+def _flip_da_indexers(entry, da_indexers) -> bool:
```

Comment:
nit: Maybe call it `_flip_da_indexers_y`, might be a clearer name? I don't know. Not a big deal and I'm not sure we'd ever have need for an x-axis version. Just a thought, take it or leave it.